security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -553,9 +557,18 @@ func (sc *SecretCache) rotate(updateRootFlag bool) {
 			// Send the notification to close the stream if token is expired, so that client could re-connect with a new token.
 			if sc.isTokenExpired() {
 				cacheLog.Debugf("%s token expired", logPrefix)
-				sc.callbackWithTimeout(connKey, nil /*nil indicates close the streaming connection to proxy*/)
-
-				return true
+				if fetchToken != nil {
+					t, err := fetchToken()
+					if err != nil {
+						cacheLog.Errorf("failed to fetch new token, closing connection: %v", err)
+						sc.callbackWithTimeout(connKey, nil /*nil indicates close the streaming connection to proxy*/)
+						return true
+					}
+					e.Token = t
```

Comment:
I think you still need to call sc.callbackWithTimeout() with the updated secret, right?